prow/config/agent.go

```diff
@@ -40,46 +40,54 @@ type Agent struct {
 	subscriptions []DeltaChan
 }
 
+func lastConfigModTime(prowConfig, jobConfig string) (time.Time, error) {
+	// Check if the file changed to see if it needs to be re-read.
+	// os.Stat follows symbolic links, which is how ConfigMaps work.
+	prowStat, err := os.Stat(prowConfig)
+	if err != nil {
+		logrus.WithField("prowConfig", prowConfig).WithError(err).Error("Error loading prow config.")
+		return time.Time{}, err
+	}
+	recentModTime := prowStat.ModTime()
+	// TODO(krzyzacy): allow empty jobConfig till fully migrate config to subdirs
+	if jobConfig != "" {
+		jobConfigStat, err := os.Stat(jobConfig)
+		if err != nil {
+			logrus.WithField("jobConfig", jobConfig).WithError(err).Error("Error loading job configs.")
+			return time.Time{}, err
+		}
+
+		if jobConfigStat.ModTime().After(recentModTime) {
+			recentModTime = jobConfigStat.ModTime()
+		}
+	}
+	return recentModTime, nil
+}
+
 // Start will begin polling the config file at the path. If the first load
 // fails, Start will return the error and abort. Future load failures will log
 // the failure message but continue attempting to load.
 func (ca *Agent) Start(prowConfig, jobConfig string) error {
+	lastModTime, err := lastConfigModTime(prowConfig, jobConfig)
+	if err != nil {
+		lastModTime = time.Now()
```

Comment:
Since our intent is to always load in this case, could we use the zero time?

Reply:
Thank you for the review. Done.